### DISCUSSION TITLE
useGenericAuth: Any way to access executionArgs.rootValue?

### DISCUSSION THREAD

**Discussion by robertherber** [dotansimha/envelop#1752] [Q&A]
When using useGenericAuth I'd like to be able to access the rootValue being passed to the resolver, but it is always undefined. I'm guessing it's because it's executed before the resolver.

The use case is as follows (I'd like to have access to the groups `rootValue._id` when determining access to `aProtectedResolver`):
```graphql
type Group {
  _id: ObjectId!
  name: String!
  aProtectedResolver: SomeResult! @auth(requiredGroupRole: ADMIN)
}
```

Not sure about the lifecycle here - I would assume this would make most sense - but I guess it's not how it works today:
1. @auth validation runs for the rootQuery (let's say groupById)
2. groupById runs if validation passes
3. @auth validation runs for aProtectedResolver and other field resolvers (**with rootValue passed from the rootQuery** <- apparently this doesn't seem to happen)
4. aProtectedResolver runs if validation passes

Is accessing rootValue possible in useGenericAuth somehow? Am I misunderstanding the flow of things?

**Comment by robertherber** [comment]
And the same issue seems to go for `executionArgs.variableValues`, where I'm not getting undefined but an empty object every single time.

**Reply by ardatan** [reply]
`rootValue` is different than `root` parameter of the resolver function.
`variableValues` is different than `args` paramater of the resolver function.

Validation is done before the execution happens. So the invocation of resolver functions are not there yet.

**Reply by robertherber** [reply]
[In this example](https://the-guild.dev/graphql/envelop/plugins/use-generic-auth#with-a-custom-validation-function-per-field) both gets passed to the auth extension of the resolver, which shows passing these exact parameters and using them there.

Is that example even working? From my understanding rootValue and variableValues will always be empty when calling the auth extension.
